lib/assembly-objectfile/content_metadata.rb

```diff
@@ -152,6 +155,13 @@ def self.create_content_metadata(params = {})
                 resource_type_description = resource_has_non_images ? resource_type_descriptions[:object] : resource_type_descriptions[:book]
               when :map
                 resource_type_description = resource_type_descriptions[:map]
+              when :'3d'
+                 resource_extensions = resource_files.collect {|obj| obj.ext}
+                 if (resource_extensions & VALID_THREE_DIMENSION_EXTENTIONS).empty? # if this resource contains no known 3D file extensions, the resource type is file
```

Comment:
why is there no clause like this for any of the other style values?  This doesn't imply, as it does for book, that it is really an "object" instead of a "book"?

Reply:
the 3d content metadata spec (https://consul.stanford.edu/display/chimera/DOR+content+types,+resource+types+and+interpretive+metadata) calls for 3d files to be of resource type="3d" and anything else will be of resource type="file"...there are similar types of checks for book (though they go to "object" instead of "file" resource types) but the code looks a little different (it uses the ternary operator instead of an if/then clause)

see https://argo.stanford.edu/view/druid:fx993tk1331 for an existing 3d object like this